hw/ip/tlul/generic_dv/xbar_tests.hjson

```diff
@@ -100,7 +100,7 @@
                  // Large delays can cause this test to time out for large crossbar configurations.
                  // Thus reduce the number of runs (called `num_trans` in `dv_base_vseq`) to prevent
                  // timeouts.
-                 "+max_num_trans=10",
+                 "+max_num_trans=9",
```

Comment:
I'm not sure I understand why the change in `max_num_trans` should be calculated based on the failure rate for the test. If it sometimes times out, I agree that it makes sense to have fewer transactions. But can't the new `max_num_trans` be set significantly lower to guarantee it stops ever timing out? Maybe 5 or something?

Reply:
The number of transactions is uniformly randomly selected up to `max_num_trans`. The hypothesis is that the tests that fail by timing out run with the maximum number of transactions. (I confirmed this hypothesis for a few of the seeds that failed, although not all.) If the hypothesis holds generally, reducing `max_num_trans` removes those failures.

A more 'aggressive' solution would be to reduce `max_num_trans` significantly to ensure that there never is a timeout. However, this would reduce the coverage more than necessary, so I'm not convinced we should do that.

Reply:
I think I'd forgotten the extra sampling step (picking a random number up to `max_num_trans`). Yep, this makes sense to me.